src/overlays/actors/ovl_En_Test/z_en_test.c

```diff
@@ -133,126 +204,1885 @@ static ColliderQuadInit D_808645C8 = {
+            EnTest_SetupJumpUp(this);
+        } else {
+            if ((Rand_ZeroOne() > 0.7f) && (this->actor.params != STALFOS_CEILING) &&
+                (player->swordAnimation != 0x11)) {
+                EnTest_SetupJumpBack(this);
+            } else {
+                EnTest_SetupStopAndBlock(this);
+            }
+        }
+
+        this->unk_7C8 = 8;
+    }
+}
+
+void EnTest_SetupStunned(EnTest* this) {
+    this->unk_7C8 = 0xB;
+    this->unk_7DE = 0;
+    this->swordState = STALFOS_SWORD_OFF;
+    this->skelAnime.playSpeed = 0.0f;
+    this->actor.speedXZ = -4.0f;
+
+    if (this->lastDamageEffect == STALFOS_DMGEFF_LIGHT) {
+        Actor_SetColorFilter(&this->actor, -0x8000, 0x78, 0, 0x50);
+    } else {
+        Actor_SetColorFilter(&this->actor, 0, 0x78, 0, 0x50);
+
+        if (this->lastDamageEffect == STALFOS_DMGEFF_FREEZE) {
+            this->iceTimer = 36;
+        } else {
+            Animation_PlayOnceSetSpeed(&this->skelAnime, &gStalfosFlinchFromHitFrontAnim, 0.0f);
+        }
+    }
+
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    EnTest_SetupAction(this, EnTest_Stunned);
+}
+
+void EnTest_Stunned(EnTest* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 1.0f, 0.0f);
+
+    if (this->actor.colorFilterTimer == 0) {
+        if (this->actor.colChkInfo.health == 0) {
+            func_80862FA8(this, globalCtx);
+            return;
+        }
+
+        if (player->swordState != 0) {
+            if ((this->actor.bgCheckFlags & 8) &&
+                ((ABS((s16)(this->actor.wallYaw - this->actor.shape.rot.y)) < 0x38A4) &&
+                 (this->actor.xzDistToPlayer < 80.0f))) {
+                EnTest_SetupJumpUp(this);
+            } else {
+                if ((Rand_ZeroOne() > 0.7f) && (player->swordAnimation != 0x11)) {
+                    EnTest_SetupJumpBack(this);
+                } else {
+                    EnTest_SetupStopAndBlock(this);
+                }
+            }
+
+            this->unk_7C8 = 8;
+
+        } else {
+            this->actor.speedXZ = 0.0f;
+            if (!EnTest_ReactToProjectile(globalCtx, this)) {
+                EnTest_ChooseAction(this, globalCtx);
+            }
+        }
+    }
+}
+
+void EnTest_SetupSidestepAgro(EnTest* this, GlobalContext* globalCtx) {
+    if (Actor_OtherIsTargeted(globalCtx, &this->actor)) {
+        EnTest_SetupSidestepInactive(this);
+        return;
+    }
+
+    Animation_MorphToLoop(&this->skelAnime, &gStalfosSidestepAnim, -2.0f);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 0xFA0, 1);
+    this->actor.speedXZ = (globalCtx->gameplayFrames & 1) ? -4.0f : 4.0f;
+    this->actor.world.rot.y = this->actor.shape.rot.y + 0x3FFF;
+    this->timer = (Rand_ZeroOne() * 20.0f) + 20.0f;
+    this->unk_7C8 = 0x18;
+    EnTest_SetupAction(this, EnTest_SidestepAgro);
+    this->unk_7EC = 0.0f;
+}
```

Comment:
Feel free to ignore this suggestion if you don't like it.

```suggestion
void EnTest_SetupSidestepAgro(EnTest* this, GlobalContext* globalCtx) {
    if (Actor_OtherIsTargeted(globalCtx, &this->actor)) {
        EnTest_SetupSidestepInactive(this);
    } else {
        Animation_MorphToLoop(&this->skelAnime, &gStalfosSidestepAnim, -2.0f);
        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 0xFA0, 1);
        this->actor.speedXZ = (globalCtx->gameplayFrames & 1) ? -4.0f : 4.0f;
        this->actor.world.rot.y = this->actor.shape.rot.y + 0x3FFF;
        this->timer = (Rand_ZeroOne() * 20.0f) + 20.0f;
        this->unk_7C8 = 0x18;
        EnTest_SetupAction(this, EnTest_SidestepAgro);
        this->unk_7EC = 0.0f;
    }
}
```

Reply:
Both work well. This actor in general has alot of early returns cause of early action changes. This one only happens to look good cause the block beneath it isnt that large. Ill prob leave it as is